content/kubermatic/master/installation/start_kkp/guides/setup_repository/_index.en.md

```diff
@@ -0,0 +1,220 @@
+
+Use `gcloud` CLI tool to create these credentials.
+
+```bash
+export GCLOUD_PROJECT="{{ .Configuration.KubernetesSpec.CloudProvider.GCE.Project }}"
+export SERVICE_ACCOUNT_NAME="k1-cluster-provisioner"
+# create new service account
+gcloud iam service-accounts create "${SERVICE_ACCOUNT_NAME}"
+# get your service account id
+export SERVICE_ACCOUNT_ID=$(gcloud iam service-accounts list --filter="name:${SERVICE_ACCOUNT_NAME}" --format='value(email)')
+# create policy bindings for KKP
+gcloud projects add-iam-policy-binding "${GCLOUD_PROJECT}" --member="serviceAccount:${SERVICE_ACCOUNT_ID}" --role='roles/compute.admin'
+gcloud projects add-iam-policy-binding "${GCLOUD_PROJECT}" --member="serviceAccount:${SERVICE_ACCOUNT_ID}" --role='roles/iam.serviceAccountUser'
+gcloud projects add-iam-policy-binding "${GCLOUD_PROJECT}" --member="serviceAccount:${SERVICE_ACCOUNT_ID}" --role='roles/viewer'
+gcloud projects add-iam-policy-binding "${GCLOUD_PROJECT}" --member="serviceAccount:${SERVICE_ACCOUNT_ID}" --role='roles/storage.admin'
+# create policy bindings for Google GitHub actions
+gcloud iam service-accounts add-iam-policy-binding ${SERVICE_ACCOUNT_ID} --member="serviceAccount:${SERVICE_ACCOUNT_ID}" --role='roles/iam.serviceAccountTokenCreator'
+# create a new json key for your service account
+gcloud iam service-accounts keys create --iam-account "${SERVICE_ACCOUNT_ID}" "${SERVICE_ACCOUNT_NAME}-sa-key.json"
+# check JSON file content of created service account json key, value will be used for GOOGLE_CREDENTIALS variable later
+cat "${SERVICE_ACCOUNT_NAME}-sa-key.json"
+```
+
+See [KubeOne documentation](https://docs.kubermatic.com/kubeone/master/architecture/requirements/machine_controller/google_cloud/gcp/) for more details.
+{{% /tab %}}
+{{% tab name="OpenStack" %}}
+Terraform provider will need to set the environment variables to access the OpenStack API.
+
+See there related KubeOne [documentation](https://docs.kubermatic.com/kubeone/master/guides/credentials/#environment-variables), select OpenStack tab.
+{{% /tab %}}
+{{% tab name="vSphere" %}}
+Terraform provider will need to set the environment variables to access the vSphere instance.
+
+See there related KubeOne [documentation](https://docs.kubermatic.com/kubeone/master/guides/credentials/#environment-variables), select vSphere tab and
+pay attention to the vSphere specific [permissions](https://docs.kubermatic.com/kubeone/master/architecture/requirements/machine_controller/vsphere/vsphere/).
+{{% /tab %}}
+{{% /tabs %}}
+
+## Generate SSH keys
+
+SSH public/private key-pair is used for accessing the cluster nodes. You can generate these keys locally, and you will need to set them inside the Pipeline Variables below.
+
+You can use following command to generate the keys:
+```bash
+ssh-keygen -t rsa -b 4096 -C "admin@kubermatic.com" -f ~/.ssh/k8s_rsa
+```
+
+## Setup Pipeline Secrets / Variables
+
+{{< tabs name="Pipeline Variables" >}}
+{{% tab name="GitHub + AWS" %}}
```

Comment:
Do we need a tab for each Git Provider/Cloud Provider combination? The environment variables should be the same across Git Providers, so I think we can just mention how to add secrets for each Git Provider at the beginning, and then have tabs only for each cloud provider.